flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -590,6 +604,15 @@ public Execution resetForNewExecution(final long timestamp, final long originati
 
 				this.currentExecution = newExecution;
 
+				synchronized (this.inputSplits){
+					for (InputSplit split: this.inputSplits){
+						if (split != null) {
```

Comment:
If we only add non-null splits, we could drop the check here. Would that make more sense?

Reply:
Agree. However this loop has been replaced with the new `Collection<InputSplit>` parameter of `InputSplitAssigner.returnInputSplit()`.